sr_port/op_zhalt.c

```diff
@@ -57,7 +58,8 @@ void op_zhalt(mval *returncode)
 #	endif
 	if (0 < TREF(gtmci_nested_level))
 	{	/* Need to return to caller - not halt (halting out of this call-in level) */
-		mumps_status = retcode;
+		mumps_status = SUCCESS;
+		TREF(gtmci_retval) = returncode;
```

Comment:
I don't think there is an issue here. The window for a problem if one were to exist is between op_halt() setting TREF(gtmci_retval) and when it calls op_zg1() which calls goframes() which unwinds the stack and when it gets to the first frame, it would copy TREF(gtmci_retval) contents to another mval that is the actual return mval which is protected since it lives in a mv_stent so as far as I'm able to see, the window is across 3 C modules that don't do anything that affects the stringpool so nothing should be triggered to cause an issue and there's not likely to be anything either. I can put a comment to this effect somewhere if you think that would clarify anything.

Reply:
In that case, can we add a DBG_MARK_STRINGPOOL_UNUSABLE/DBG_MARK_STRINGPOOL_USABLE macro pair for that time window that way we ensure no stringpool usages ever happen now or in the future.